### ISSUE TITLE
Add "Select Network" screen after you create/import a new wallet

### ISSUE THREAD

**Issue by colourfreak** [AlphaWallet/alpha-wallet-android#2618]
**Problem:**
During user tests in Amsterdam, I found that users don't really know how to enable their favorite networks. They don't even understand that we support multiple networks.

**Solution:**
Add one more screen where they can select multiple networks from the existing base (we make a long list of all 32 supported networks). If they would need a custom one, then they could do this later.

**Why?**

- first, we let users know we support multiple chains
- the wallet is already customized, so they don't have to go to the Settings tab

Invision Prototype:
https://invis.io/RA12NU40P8SW

<img width="899" alt="Screen Shot 2022-05-23 at 15 36 31" src="https://user-images.githubusercontent.com/51817359/169831663-57ce6126-0ae0-44ae-addf-27e717c2b694.png">

**Zeplin:**
https://zpl.io/kDNl9OW
https://zpl.io/6NRPwZg

**Comment by colourfreak** [comment]
[#4507](https://github.com/AlphaWallet/alpha-wallet-ios/issues/4507)

**Comment by colourfreak** [comment]
https://user-images.githubusercontent.com/51817359/170224901-cb01c3c1-709d-4c8a-aa53-dd7043875296.mov

**Comment by JamesSmartCell** [comment]
Hi @colourfreak, I like this idea; can you just confirm:

1. It only happens when creating/importing the first wallet? how about we also have this occur once when user creates/imports, if they have not ever opened this dialog?
2. If user has selected some mainnet chains, and some testnet chains, when they click continue, only the mainnet chains will be selected. However once they switch to testnet, the chains they selected previously will now be active.

**Comment by colourfreak** [comment]
> Hi @colourfreak, I like this idea; can you just confirm:
> 
> 1. It only happens when creating/importing the first wallet? how about we also have this occur once when user creates/imports, if they have not ever opened this dialog?
> 2. If user has selected some mainnet chains, and some testnet chains, when they click continue, only the mainnet chains will be selected. However once they switch to testnet, the chains they selected previously will now be active? Because as you know (you designed it) we only show mainnet or testnet.

1. We should prompt users with network selection flow every time they create or import a wallet. If you have 10 wallets imported already, the number 11th will also have this. 

2. Yes I designed. The main reason behind modes was that we were facing scams at that time. But things got changed and people are using testnets in parallel to mainnets. You can mint your tickets on Rinkeby and you save money. 

I made the issue for iOS already, I am going to make one for Android too. 
https://github.com/AlphaWallet/alpha-wallet-ios/issues/4663

**Comment by JamesSmartCell** [comment]
> > Hi @colourfreak, I like this idea; can you just confirm:
> > 
> > 1. It only happens when creating/importing the first wallet? how about we also have this occur once when user creates/imports, if they have not ever opened this dialog?
> > 2. If user has selected some mainnet chains, and some testnet chains, when they click continue, only the mainnet chains will be selected. However once they switch to testnet, the chains they selected previously will now be active? Because as you know (you designed it) we only show mainnet or testnet.
> 
> 1. We should prompt users with network selection flow every time they create or import a wallet. If you have 10 wallets imported already, the number 11th will also have this.
> 2. Yes I designed. The main reason behind modes was that we were facing scams at that time. But things got changed and people are using testnets in parallel to mainnets. You can mint your tickets on Rinkeby to save money.
> 
> I made the issue for iOS already, I am going to make one for Android too. [AlphaWallet/alpha-wallet-ios#4663](https://github.com/AlphaWallet/alpha-wallet-ios/issues/4663)

ok got it, makes sense.

Should we make testnets more obvious somehow too?

**Comment by JamesSmartCell** [comment]
Oh one last thing - can a user pick different networks for each wallet?

Because, some wallets I have are testnet only, some wallets are mainnet only and some are both.

It would be really nice if I could pick which networks get shown for different wallets. Or do you think this is too complicated?

**Comment by colourfreak** [comment]
> Oh one last thing - can a user pick different networks for each wallet?
> 
> Because, some wallets I have are testnet only, some wallets are mainnet only and some are both.
> 
> It would be really nice if I could pick which networks get shown for different wallets. Or do you think this is too complicated?

That would be great.

**Comment by seabornlee** [comment]
@hboon Should this be included in Phase one?

**Comment by hboon** [comment]
Yes, but only because iOS already did it earlier. So phase 1, but not high priority.

**Comment by seabornlee** [comment]
> Yes, but only because iOS already did it earlier before we defined phases. So phase 1, but not high priority.

Ok.

**Comment by hboon** [comment]
Has this be implemented? Let's undo it if it is.

**Comment by hboon** [comment]
> Has this be implemented? Let's undo it if it is.

Refer to https://github.com/AlphaWallet/alpha-wallet-ios/issues/5982

Buzz @seabornlee @JamesSmartCell

**Comment by seabornlee** [comment]
Yes, the implementation is complete. Before I proceed with any potential reversion, could you please provide a reason or justification for doing so?

**Comment by hboon** [comment]
> Yes, the implementation is complete. Before I proceed with any potential reversion, could you please provide a reason or justification for doing so?

Main reason being: it adds more friction for user to get started. It's one more thing for them to decide and do before they can do anything with the app. We ought to have the best defaults (chains enabled) for them.

If we want to drive usage and help them be more familiar with the app, we ought to get them to send a transaction instead (we probably will).

**Comment by seabornlee** [comment]
Thanks! I understand your point. For beginners, adding more cognitive load can be overwhelming and may hinder their learning process. It's important to start with simpler concepts and gradually introduce more complex topics as they gain confidence and experience.
